presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -115,9 +117,17 @@ public Stripe readStripe(StripeInformation stripe, AggregatedMemoryContext syste
             if (includedOrcColumns.contains(stream.getColumn())) {
                 streams.put(new StreamId(stream), stream);
 
-                ColumnEncodingKind columnEncoding = columnEncodings.get(stream.getColumn()).getColumnEncodingKind();
-                if (columnEncoding == DICTIONARY && stream.getStreamKind() == StreamKind.IN_DICTIONARY) {
-                    hasRowGroupDictionary = true;
+                if (stream.getStreamKind() == StreamKind.IN_DICTIONARY) {
+                    ColumnEncoding columnEncoding = columnEncodings.get(stream.getColumn());
+
+                    hasRowGroupDictionary |= columnEncoding.getColumnEncodingKind() == DICTIONARY;
```

Comment:
instead of using `|=`, we can just use if +  assignment -- I find myself confused this time reading `|=` :
```java
                    if (columnEncoding.getColumnEncodingKind() == DICTIONARY) {
                        hasRowGroupDictionary = true;
                    }

                    Optional<List<DwrfSequenceEncoding>> additionalSequenceEncodings = columnEncoding.getAdditionalSequenceEncodings();
                    if (additionalSequenceEncodings.isPresent()
                            && additionalSequenceEncodings.get().stream()
                            .map(DwrfSequenceEncoding::getValueEncoding)
                            .anyMatch(encoding -> encoding.getColumnEncodingKind() == DICTIONARY)) {
                        hasRowGroupDictionary = true;
                    }
```